### PR TITLE
Angular envelope implementation for the example Ping Pong View

### PR DESCRIPTION
**Pull request by thiagoelg**
# In this PR
- Refactor `ping-pong-view` package to remove most of its view layer, leaving only the necessary to implement a Ping Pong View wherever needed;
- Small refactor to `ping-pong-view-react` to use the refactored `ping-pong-view`, completing it with the removed components from `ping-pong-view` and adapting it to the new API;
- Added hooks and new functions to the Ping Pong example in the `webapp` package, allowing for easier integration and removing repeated code;
- Standardize CSS across Ping Pong View implementations;
- Added Ping Pong View implementation developed in Angular 13;
- Added a new "Ping-Pong Mixed Views" page in `webapp` to showcase mixed implementations of the Ping Pong View working together.
- Update CI node version to `16.13.2`.
- Update Typescript and TSLib versions to `^4.4.2` and `^2.3.1` respectively.

## Showcase:

- Ping Pong View implemented in React, running inside an iFrame
  
  https://user-images.githubusercontent.com/10515135/150586796-4d1aa8a3-dba4-4dfd-9820-ae8f925b95f8.mp4

- Ping Pong View implemented in React, running inside a Div
  
  https://user-images.githubusercontent.com/10515135/150586956-51fcd12b-4fc9-4460-8ea8-b31090f649b1.mp4

- Ping Pong View implemented in Angular, running inside an iFrame
  
  https://user-images.githubusercontent.com/10515135/150587079-4f9e072e-b5f5-4d9c-b8cf-2dd61f1f3dc2.mp4

- Ping Pong View implemented in Angular, built as a Web Component, running inside a Div
  
  https://user-images.githubusercontent.com/10515135/150587093-b268fc6b-ab0f-4f50-826d-c78d5bcb2f2e.mp4

- Ping Pong Mixed Views (all implementations)
  
  https://user-images.githubusercontent.com/10515135/150587099-e3a7d60f-fe24-42d5-b2f0-5043ac99dd0a.mp4

### CODE REVIEW

File: examples/ping-pong-view-angular/.gitignore [hunk 1]
```diff
@@ -0,0 +1 @@
+.angular
```
Comment (tiagobento) [inline examples/ping-pong-view-angular/.gitignore]: We don't have nested .gitignores in this repo.. Can you please add it to the root one?

**Review by tiagobento — APPROVED** [review]
@thiagoelg Excellent stuff!!!!! Just a minor comment regarding `.gitignore`s.

**Comment by manstis** [comment]
Oh no, a nosy blighter has seen this PR and wonders...

- What is a "Ping-Pong View Envelope"?
- What use-case purpose does it serve?
- Do I need to know about it (for the PMML editor)?

**Comment by tiagobento** [comment]
@manstis Hi Michael! What a delight to see you here 😄 

Ping Pong View is an *example* envelope we created to showcase the capabilities of the Multiplying Architecture without the complexity and history that our Editor envelope has. @ljmotta Wrote a series of blogposts showcasing what the Multiplying Architecture can do for your components, and Ping Pong View was the most complete example of all.

Following a similar approach that we have for the Editors (where the Envelope specifies an API and allows us to have multiple implementations of it, independent of language).

The examples webapp (at `examples/webapp`) contains all the webapp related examples we could gather, and this PR introduces a new Angular-based implementation of the Ping Pong View.

We've been wanting to do this for a while, but didn't find the time to do it. With @thiagoelg joining the team bringing some Angular bagage with him, we found the perfect match. Now we have examples of Envelopes with React, GWT, and Angular components inside. All transparent for the Channels consuming these Envelopes.

##### Answering your questions:
_What is a "Ping-Pong View Envelope"?_
I guess this one is already exaplained.

_What use-case purpose does it serve?_
It's just an example. A showcase of the Multiplying Architecture capabilities.

_Do I need to know about it (for the PMML editor)?_
No, but you can certainly get familiar with it to steal ideas and for inspiration. Even to improve the current Editor API that we have, as it's been stable now for some time :)

**Comment by tiagobento** [comment]
@thiagoelg Can you please update the README with the new version of Node? Also, maybe we could bump it to 16.13.2?

**Pull request merged**